src/handlers/poster-base/index.tsx

```diff
@@ -68,25 +82,29 @@ export default function PosterView() {
     }
   }, [posters, loading, posterTimeout]);
 
-  const selectedPoster = posters && posters.length > 0 && posterIndex !== undefined ? posters[posterIndex] : null;
+  const selectedPoster = posters && posters.length > 0 && posterIndex !== undefined ? posters[posterIndex] : undefined;
 
   return (
     <div
       className="h-screen w-screen bg-center bg-cover bg-no-repeat"
       style={{ backgroundImage: 'url("base/poster-background.png")' }}
     >
       <div className="overflow-hidden w-full h-full">
-        <PosterCarousel posters={posters || []} currentPoster={!posterIndex ? 0 : posterIndex} setTitle={setTitle} />
-        <ProgressBar
-          title={title}
-          seconds={posterTimeout !== undefined ? selectedPoster?.timeout : undefined}
-          posterIndex={posterIndex}
-          minimal={selectedPoster?.footer === FooterSize.MINIMAL}
-          hide={selectedPoster?.footer === FooterSize.HIDDEN}
-          borrelMode={borrelMode}
-          nextPoster={nextPoster}
-          pausePoster={pausePoster}
+        <PosterCarousel
+          posters={posters || []}
+          currentPoster={!posterIndex ? 0 : posterIndex}
+          setTitle={setTitle}
+          localPosterRenderer={localPosterRenderer}
         />
+        {overlay({
+          poster: selectedPoster,
+          posterTitle: title,
+          seconds: posterTimeout !== undefined ? selectedPoster?.timeout : undefined,
+          posterIndex: posterIndex,
+          nextPoster: nextPoster,
+          pausePoster: pausePoster,
+          borrelMode: borrelMode,
+        })}
```

Comment:
Is it not possible to make this a `{children}` prop? Then you don't have to pass complete components as attributes, which looks a bit ugly to me in `/poster-gewis/index.tsx` and in the Hubble equivalent.

Reply:
Can you not use `React.memo` to prevent this?

Reply:
No because the overlay _should_ be rendered right? Because as the props of the overlay change, then the overlay does need a re-render, for example in case of a new poster that a minimal footer?